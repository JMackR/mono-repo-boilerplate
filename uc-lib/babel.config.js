module.exports = {
  presets: [["@babel/preset-env", { targets: { node: "current" } }], "module:metro-react-native-babel-preset"],
  plugins: ["@babel/plugin-proposal-class-properties", "inline-json-import"],
  sourceType: "unambiguous",
}
