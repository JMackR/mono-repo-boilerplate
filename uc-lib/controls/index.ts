export * from "./image"
export * from "./button"
export * from "./text"
export * from "./activity-indicator"
export * from "./separator"
export * from "./list"
export * from "./section-list"
export * from "./badge"
export * from "./toggle"
export * from "./login-button"
export * from "./layout"
export * from "./background"
export * from "./grid"
export * from "./touchable"
export * from "./view-block"
export * from "./text-entry"
export * from "./scroll-view"
export * from "./circle"
export * from "./slider"
export * from "./clickable"
export * from "./map"
export * from "./visibility-check"
export * from "./strike-through-text"
// export * from "./stripe-text-entry"
export * from "./input"
