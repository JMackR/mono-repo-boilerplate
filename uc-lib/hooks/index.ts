export * from "./use-infinite-scroll"
export * from "./selectable"
export * from "./screen-provider"
export * from "./loading"
