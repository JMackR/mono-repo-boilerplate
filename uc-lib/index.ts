export * from "./animations"
export * from "./assets"
export * from "./controls"
export * from "./themes"
export * from "./widgets"
export * from "./utilities"
export * from "./hooks"
export * from "./validators"
