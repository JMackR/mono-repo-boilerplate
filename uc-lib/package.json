{
  "name": "redibs-ucl",
  "version": "0.1.0",
  "private": true,
  "scripts": {
    "test": "jest --projects test/react test/react-native test/non-react"
  },
  "dependencies": {
    "@types/ramda": "^0.27.14",
    "classnames": "2.2.6",
    "google-map-react": "1.1.7",
    "hoist-non-react-statics": "3.3.2",
    "imask": "6.0.5",
    "react-compound-slider": "2.5.0",
    "react-native-autolink": "3.0.0",
    "react-native-fs": "^2.16.6",
    "zxcvbn": "4.4.2"
  },
  "devDependencies": {},
  "prettier": {
    "printWidth": 125,
    "semi": false,
    "singleQuote": false,
    "trailingComma": "all"
  },
  "jest-junit": {
    "outputDirectory": "../build",
    "ancestorSeparator": " â€º ",
    "classNameTemplate": "{classname}",
    "suiteName": "redibs-redibs-ucl jest tests",
    "suiteNameTemplate": "{filename}",
    "titleTemplate": "{title}",
    "uniqueOutputName": "true"
  }
}
