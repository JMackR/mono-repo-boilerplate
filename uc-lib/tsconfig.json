{
  "extends": "../tsconfig.json",
  "compilerOptions": {
    "baseUrl": ".",
    "outDir": "lib",
    "rootDir": "../",
    "jsx": "react-native",
    "composite": true,
    "declarationMap": true,
    "isolatedModules": true,
    "sourceMap": true,
    "moduleResolution": "Node",
    "esModuleInterop": true,
    "typeRoots": ["../node_modules/@types", "./types"]
  },
  "include": ["./"],
  "exclude": ["node_modules", "babel.config.js", "metro.config.js", "jest.config.js", "types"]
}
