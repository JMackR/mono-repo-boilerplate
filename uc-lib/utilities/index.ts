export * from "./dimensions-utility"
export * from "./media-query"
