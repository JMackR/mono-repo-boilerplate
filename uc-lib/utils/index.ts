export * from "./iif"
export * from "./format-number"
