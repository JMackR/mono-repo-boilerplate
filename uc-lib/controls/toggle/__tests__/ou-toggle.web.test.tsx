import React from "react"
import { shallow, ShallowWrapper } from "enzyme"
import { Toggle } from "uc-lib"
import * as renderer from "react-test-renderer"
import { StyleSheetTestUtils } from "aphrodite/no-important"
import { createSerializer } from "jest-aphrodite-react/no-important"
import Switch from "react-switch"

expect.addSnapshotSerializer(createSerializer(() => StyleSheetTestUtils, { removeVendorPrefixes: true }))

describe("Toggle Functionality Tests", () => {
  let wrapper: ShallowWrapper
  const click = jest.fn()
  beforeEach(() => {
    wrapper = shallow(<Toggle state={false} onChange={click} />)
  })
  test("Click calls passed in onChange function", () => {
    wrapper
      .find(Switch)
      .props()
      .onChange(true, {} as MouseEvent, "test")
    expect(click).toHaveBeenCalledTimes(1)
  })
})

describe("Toggle Snapshot Tests", () => {
  test("Toggle with on state renderer.creates properly", () => {
    const toggle = renderer.create(<Toggle state={true} onChange={jest.fn()} />)
    expect(toggle).toMatchSnapshot()
  })
  test("Toggle with off state renderer.creates properly", () => {
    const toggle = renderer.create(<Toggle state={false} onChange={jest.fn()} />)
    expect(toggle).toMatchSnapshot()
  })
  test("Toggle with off state renderer.creates properly", () => {
    const toggle = renderer.create(<Toggle state={false} onChange={jest.fn()} disabled={true} />)
    expect(toggle).toMatchSnapshot()
  })
})
