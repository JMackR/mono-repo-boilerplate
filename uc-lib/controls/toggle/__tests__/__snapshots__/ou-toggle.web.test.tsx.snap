// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle Snapshot Tests Toggle with off state renderer.creates properly 1`] = `
<label>
  <div
    style={
      Object {
        "MozTransition": "opacity 0.25s",
        "MozUserSelect": "none",
        "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
        "WebkitTransition": "opacity 0.25s",
        "WebkitUserSelect": "none",
        "borderRadius": 15,
        "direction": "ltr",
        "display": "inline-block",
        "msUserSelect": "none",
        "opacity": 1,
        "position": "relative",
        "textAlign": "left",
        "touchAction": "none",
        "transition": "opacity 0.25s",
        "userSelect": "none",
      }
    }
  >
    <div
      className="react-switch-bg"
      onClick={[Function]}
      onMouseDown={[Function]}
      style={
        Object {
          "MozTransition": "background 0.25s",
          "WebkitTransition": "background 0.25s",
          "background": "#ABABAB",
          "borderRadius": 15,
          "cursor": "pointer",
          "height": 30,
          "margin": 0,
          "position": "relative",
          "transition": "background 0.25s",
          "width": 57,
        }
      }
    />
    <div
      className="react-switch-handle"
      onClick={[Function]}
      onMouseDown={[Function]}
      onTouchCancel={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "MozTransition": "background-color 0.25s, transform 0.25s, box-shadow 0.15s",
          "WebkitTransition": "background-color 0.25s, transform 0.25s, box-shadow 0.15s",
          "background": "#FFFFFF",
          "border": 0,
          "borderRadius": "50%",
          "boxShadow": null,
          "cursor": "pointer",
          "display": "inline-block",
          "height": 24,
          "outline": 0,
          "position": "absolute",
          "top": 3,
          "transform": "translateX(3px)",
          "transition": "background-color 0.25s, transform 0.25s, box-shadow 0.15s",
          "width": 24,
        }
      }
    />
    <input
      checked={false}
      disabled={false}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyUp={[Function]}
      role="switch"
      style={
        Object {
          "border": 0,
          "clip": "rect(0 0 0 0)",
          "height": 1,
          "margin": -1,
          "overflow": "hidden",
          "padding": 0,
          "position": "absolute",
          "width": 1,
        }
      }
      type="checkbox"
    />
  </div>
</label>
`;

exports[`Toggle Snapshot Tests Toggle with off state renderer.creates properly 2`] = `
<label>
  <div
    style={
      Object {
        "MozTransition": "opacity 0.25s",
        "MozUserSelect": "none",
        "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
        "WebkitTransition": "opacity 0.25s",
        "WebkitUserSelect": "none",
        "borderRadius": 15,
        "direction": "ltr",
        "display": "inline-block",
        "msUserSelect": "none",
        "opacity": 0.5,
        "position": "relative",
        "textAlign": "left",
        "touchAction": "none",
        "transition": "opacity 0.25s",
        "userSelect": "none",
      }
    }
  >
    <div
      className="react-switch-bg"
      onClick={null}
      onMouseDown={[Function]}
      style={
        Object {
          "MozTransition": "background 0.25s",
          "WebkitTransition": "background 0.25s",
          "background": "#ABABAB",
          "borderRadius": 15,
          "cursor": "default",
          "height": 30,
          "margin": 0,
          "position": "relative",
          "transition": "background 0.25s",
          "width": 57,
        }
      }
    />
    <div
      className="react-switch-handle"
      onClick={[Function]}
      onMouseDown={null}
      onTouchCancel={null}
      onTouchEnd={null}
      onTouchMove={null}
      onTouchStart={null}
      style={
        Object {
          "MozTransition": "background-color 0.25s, transform 0.25s, box-shadow 0.15s",
          "WebkitTransition": "background-color 0.25s, transform 0.25s, box-shadow 0.15s",
          "background": "#FFFFFF",
          "border": 0,
          "borderRadius": "50%",
          "boxShadow": null,
          "cursor": "default",
          "display": "inline-block",
          "height": 24,
          "outline": 0,
          "position": "absolute",
          "top": 3,
          "transform": "translateX(3px)",
          "transition": "background-color 0.25s, transform 0.25s, box-shadow 0.15s",
          "width": 24,
        }
      }
    />
    <input
      checked={false}
      disabled={true}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyUp={[Function]}
      role="switch"
      style={
        Object {
          "border": 0,
          "clip": "rect(0 0 0 0)",
          "height": 1,
          "margin": -1,
          "overflow": "hidden",
          "padding": 0,
          "position": "absolute",
          "width": 1,
        }
      }
      type="checkbox"
    />
  </div>
</label>
`;

exports[`Toggle Snapshot Tests Toggle with on state renderer.creates properly 1`] = `
<label>
  <div
    style={
      Object {
        "MozTransition": "opacity 0.25s",
        "MozUserSelect": "none",
        "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
        "WebkitTransition": "opacity 0.25s",
        "WebkitUserSelect": "none",
        "borderRadius": 15,
        "direction": "ltr",
        "display": "inline-block",
        "msUserSelect": "none",
        "opacity": 1,
        "position": "relative",
        "textAlign": "left",
        "touchAction": "none",
        "transition": "opacity 0.25s",
        "userSelect": "none",
      }
    }
  >
    <div
      className="react-switch-bg"
      onClick={[Function]}
      onMouseDown={[Function]}
      style={
        Object {
          "MozTransition": "background 0.25s",
          "WebkitTransition": "background 0.25s",
          "background": "#00A87E",
          "borderRadius": 15,
          "cursor": "pointer",
          "height": 30,
          "margin": 0,
          "position": "relative",
          "transition": "background 0.25s",
          "width": 57,
        }
      }
    />
    <div
      className="react-switch-handle"
      onClick={[Function]}
      onMouseDown={[Function]}
      onTouchCancel={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "MozTransition": "background-color 0.25s, transform 0.25s, box-shadow 0.15s",
          "WebkitTransition": "background-color 0.25s, transform 0.25s, box-shadow 0.15s",
          "background": "#FFFFFF",
          "border": 0,
          "borderRadius": "50%",
          "boxShadow": null,
          "cursor": "pointer",
          "display": "inline-block",
          "height": 24,
          "outline": 0,
          "position": "absolute",
          "top": 3,
          "transform": "translateX(30px)",
          "transition": "background-color 0.25s, transform 0.25s, box-shadow 0.15s",
          "width": 24,
        }
      }
    />
    <input
      checked={true}
      disabled={false}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyUp={[Function]}
      role="switch"
      style={
        Object {
          "border": 0,
          "clip": "rect(0 0 0 0)",
          "height": 1,
          "margin": -1,
          "overflow": "hidden",
          "padding": 0,
          "position": "absolute",
          "width": 1,
        }
      }
      type="checkbox"
    />
  </div>
</label>
`;
